python/src/rtree.rs

```diff
@@ -277,6 +309,44 @@ impl RTree {
         result
     }
 
+    #[classmethod]
+    pub fn from_buffer(
+        _cls: &Bound<PyType>,
+        py: Python,
+        buf: PyObject,
+    ) -> PyResult<Self> {
+        buf.extract::<RTree>(py)
+    }
+
+    pub fn to_buffer<'py>(
+        &'py self,
+        py: Python<'py>,
+    ) -> PyObject {
+        PyBytes::new_bound(py, self.0.buffer()).into()
+    }
+
+    // pre PEP 688 buffer protocol
+    #[cfg(any(Py_3_11, not(Py_LIMITED_API)))]
+    pub unsafe fn __getbuffer__(slf: PyRef<'_, Self>, view: *mut ffi::Py_buffer, flags: c_int) -> PyResult<()> {
+        let bytes = slf.0.buffer();
+        let ret = ffi::PyBuffer_FillInfo(
+            view,
+            slf.as_ptr() as *mut _,
+            bytes.as_ptr() as *mut _,
+            bytes.len().try_into().unwrap(),
+            1, // read only
+            flags,
+        );
+        if ret == -1 {
+            return Err(PyErr::fetch(slf.py()));
+        }
+        Ok(())
+    }
+    #[cfg(any(Py_3_11, not(Py_LIMITED_API)))]
+    pub unsafe fn __releasebuffer__(&self, _view: *mut ffi::Py_buffer) {
+        // is there anything to do here?
```

Comment:
I don't know. In https://github.com/kylebarron/arro3/pull/156 I hold an `Option<>` in the buffer and manually move the value out of the option to drop it. But I'm not sure if eventually Python would garbage collect the object, at which point it would be dropped